URLからアプリを起動する

alotofwe opened the pull request:
各NavigationControllerに，StoryBoardからidentifierをつけることで移動可能です．

identifierの命名規則は"HogeNavigationCongroller"で
例えば，feedはFeedNavigationControllerにしています．

URLでは，hogeの部分をhostに指定します．
例えば，feedはkakico://feedになります．

&key=valueで，URLからAppDelegateに値を渡せるようになっています．
(今はなににも使っていませんが，これでauth_tokenを渡す予定です．

Commit:
scheme://feedでフィード画面に飛ぶようにする

NavigationControllerにidentifierをつけることで移動可能
identifierの命名規則は"HogeNavigationCongroller"で
例えば，feedはFeedNavigationControllerにしている

URLでは，hogeの部分をhostに指定する

Commit:
Add URL Type

Commit:
kakiko -> kakico

Commit:
host指定なしの場合はSign inに飛ぶ

Commit:
localhost経由でSign upを叩く

メールアドレス確認の簡便のため

Commit:
空白削除

alotofwe commented:
@takuminnnn @orzup 

https://github.com/alotofwe/sample_app-1/pull/20 と共に，シミュレータでレビューをお願いします．
手順は以下のとおりです．
1. 手元のsample_app-1のブランチをaccount_activations/requestに切り替える
2. bundle exec rails s
3. iOSのブランチをhandleOpenURLに切り替える
4. シミュレータを立ち上げる
5. 通常通りSign Upを行う (urlは存在しないものでも大丈夫)
6. rails sのログにメールの文面としてactivation用URLが表示されるので，シミュレータのsafariでそこにアクセスをする
7. rails側ではactivationが完了し，iOS側ではfeed画面が立ち上がる

ku00 commented on FoodTracker/Classes/Controllers/SignupViewController.swift:
```diff
@@ -68,7 +68,8 @@ class SignupViewController: UIViewController, UITextFieldDelegate {
         SVProgressHUD.showWithMaskType(SVProgressHUDMaskType.Black)
-        Alamofire.request(.POST, "https://157.7.190.148/api/users", parameters: params, encoding: .JSON)
+        //Alamofire.request(.POST, "https://157.7.190.148/api/users", parameters: params, encoding: .JSON)
+        Alamofire.request(.POST, "http://localhost:3000/api/users", parameters: params, encoding: .JSON)
```
`(.POST, "http://localhost:3000/api/users", ~` の部分は `Router.swift` に移したいね

ku00 replied:
あとDictionaryでjsonをほげほげする部分も swiftyJSON使えばもっと楽に書けそう

alotofwe replied:
2つとも，付箋でタスクを生やしています!

ku00 replied:
ありがとう!

ku00 commented:
sign up後にfeedにすぐ飛んでるけど、これ正しい挙動なんだろうか...

alotofwe commented:
activatiom待ちページは，別タスクとして生やしているので，今は正しい挙動です~
いまはfeed以外に遷移すべきページが無い...

ku00 commented:
どんなAPIを使っているのかPRのはじめに明記しておくと親切かもしれない(一瞬なんのPRなのかわからなかった)

alotofwe commented:
これは，iOS側では新たにAPIを叩いていないんだよね...
新たにAPIを使用する時は書きます〜 :cat:

ku00 commented:
okです!!

Commit:
Router使用

Commit:
User SwiftyJSON

Commit:
Routerの構成を変更

公式ドキュメントのCRUD & Authorizationを参考にした．
API Parameter Abstractionでは，POSTの時のJSON encodeが難しかった．

ku00 commented on FoodTracker/Classes/Controllers/SignupViewController.swift:
```diff
@@ -68,19 +69,19 @@ class SignupViewController: UIViewController, UITextFieldDelegate {
+            } else{
+                var messages = json["messages"]
+                var user_errors = messages["user"]
+                SVProgressHUD.showErrorWithStatus(user_errors.dictionary!.values.first?.stringValue)
```
エラーメッセージ取るところもswiftyJSON使って簡潔に書けそう

```
json["messages"]["user"]["email"].string
```

こんな感じかな

orzup replied:
swiftyJSONよいなあ〜

ku00 commented:
リファクタリング後も、ユーザ登録(POST)できてactivationも成功してるので良さげ :+1:

ku00 commented on FoodTracker/Classes/Controllers/SignupViewController.swift:
```diff
@@ -68,19 +69,19 @@ class SignupViewController: UIViewController, UITextFieldDelegate {
+                self.presentViewController( targetViewController, animated: true, completion: nil)
+            } else{
+                var messages = json["messages"]
+                var user_errors = messages["user"]
```
messagesとuser_errorsで分けたのはなぜだろう?

Commit:
messages, user_errorsを生成しない
